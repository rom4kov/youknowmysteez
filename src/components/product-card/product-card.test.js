import { mount } from "enzyme";
import ProductCard from "./product-card.component";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

const mockStore = configureStore();
const initState = {
  product: {
    id: 1,
    imageUrl: "www.url.com",
    name: "hat",
    price: 20,
    brand: "Nike",
  },
};
const store = mockStore(initState);

it("properly render ProductCard component", () => {
  expect(
    mount(
      <Provider store={store}>
        <ProductCard />
      </Provider>
    )
  ).toMatchSnapshot();
});
