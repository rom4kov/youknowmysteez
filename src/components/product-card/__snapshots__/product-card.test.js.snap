// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`properly render ProductCard component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ProductCard>
    <styled.div>
      <div
        className="sc-dmqHEX kneRcb"
      >
        <styled.img
          alt="undefined"
        >
          <img
            alt="undefined"
            className="sc-hLseeU iwsRzB"
          />
        </styled.img>
        <styled.div>
          <div
            className="sc-eDDNvR iqCncc"
          >
            <styled.span>
              <span
                className="sc-jTrPJq ckcNaj"
              />
            </styled.span>
            <styled.span>
              <span
                className="sc-gLDzan ewZTjn"
              >
                â‚¬ 
              </span>
            </styled.span>
          </div>
        </styled.div>
        <Styled(Button)
          buttonType="inverted"
          onClick={[Function]}
        >
          <Button
            buttonType="inverted"
            className="sc-iAEyYk jvcVwh"
            onClick={[Function]}
          >
            <buttonstyles__InvertedButton
              className="sc-iAEyYk jvcVwh"
              onClick={[Function]}
            >
              <button
                className="buttonstyles__BaseButton-sc-lswy1h-0 buttonstyles__InvertedButton-sc-lswy1h-2 jJMbCT fsMtkj sc-iAEyYk jvcVwh"
                onClick={[Function]}
              >
                In den Warenkorb
              </button>
            </buttonstyles__InvertedButton>
          </Button>
        </Styled(Button)>
      </div>
    </styled.div>
  </ProductCard>
</Provider>
`;
